core/src/main/java/com/linecorp/armeria/common/CoreBlockhoundIntegration.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import java.util.ResourceBundle;
+import java.util.UUID;
+
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+import reactor.blockhound.BlockHound.Builder;
+import reactor.blockhound.integration.BlockHoundIntegration;
+
+/**
+ * A {@link BlockHoundIntegration} for the core module.
+ */
+@UnstableApi
+public final class CoreBlockhoundIntegration implements BlockHoundIntegration {
+    @Override
+    public void applyTo(Builder builder) {
+        // short locks
+        builder.allowBlockingCallsInside("com.linecorp.armeria.client.HttpClientFactory",
+                                         "pool");
+        builder.allowBlockingCallsInside("com.linecorp.armeria.internal.common.util.ReentrantShortLock",
+                                         "lock");
+
+        // Thread.yield can be eventually called when PooledObjects.copyAndClose is called
+        builder.allowBlockingCallsInside("io.netty.util.internal.ReferenceCountUpdater", "release");
+
+        // hdr histogram holds locks
+        builder.allowBlockingCallsInside("org.HdrHistogram.ConcurrentHistogram", "getCountAtIndex");
+        builder.allowBlockingCallsInside("org.HdrHistogram.WriterReaderPhaser", "flipPhase");
+
+        // StreamMessageInputStream internally uses a blocking queue
```

Comment:
It seems like it uses add instead offer? 🤔

Reply:
It seems like `LinkedBlockingQueue` internally delegates `add` to `offer`

https://github.com/openjdk/jdk/blob/f5a6b7f7c03c00c96d0055f9be31057675205e13/src/java.base/share/classes/java/util/concurrent/LinkedBlockingQueue.java#LL81C45-L81C58
https://github.com/openjdk/jdk/blob/f5a6b7f7c03c00c96d0055f9be31057675205e13/src/java.base/share/classes/java/util/AbstractQueue.java#L94

Reply:
Understood. Thanks!